### DISCUSSION TITLE
Issue when bypassing Cloudflare captcha

### DISCUSSION THREAD

**Discussion by OpsecGuy** [seleniumbase/SeleniumBase#2670] [Q&A] [answered]
Hey,
For the last few days, I do experience known behavior which is being in a Cloudflare JS challenge loop while running a test case script. I did try my own code base that uses raw driver and SB test from repo as well. Both with the same result. 
I could give more details about my raw driver version of the script, but I believe it's not needed, because even as mentioned above, a test from repo also does not bypass CF challenge.

### Chromedriver / Chrome version:
123 / 123.0.6312.106

### Selenium* versions:
selenium==4.19.0
selenium-wire==5.1.0
seleniumbase==4.25.2

### Showcase:
https://i.imgur.com/cnpCxG5.gif

### Code used from:
https://github.com/seleniumbase/SeleniumBase/blob/master/examples/raw_turnstile.py

### Tested URLs:
https://nopecha.com/demo/turnstile --> Doesn't bypass CF
https://cfcybernews.eu                        --> Doesn't bypass CF
seleniumbase.io/apps/turnstile           --> Bypass partially works (just clicks the captcha box, no real CF challenge is shown)

**Comment by mdmintz** [comment, marked as answer]
It's working fine with the right scripts. Eg:

```python
from seleniumbase import SB

with SB(uc=True, test=True) as sb:
    sb.driver.uc_open_with_reconnect("nopecha.com/demo/turnstile", 3.4)
    if sb.is_element_visible("#example-container0 iframe"):
        sb.switch_to_frame("#example-container0 iframe")
        if not sb.is_element_visible("circle.success-circle"):
            sb.driver.uc_click("span.mark", reconnect_time=3)
            sb.switch_to_frame("#example-container0 iframe")
        sb.switch_to_default_content()

    sb.switch_to_frame("#example-container5 iframe")
    sb.driver.uc_click("span.mark", reconnect_time=2.5)
    sb.switch_to_frame("#example-container5 iframe")
    sb.assert_element("svg#success-icon", timeout=3)
    sb.switch_to_parent_frame()

    if sb.is_element_visible("#example-container0 iframe"):
        sb.switch_to_frame("#example-container0 iframe")
        sb.assert_element("circle.success-circle")
        sb.switch_to_parent_frame()

    sb.set_messenger_theme(location="top_center")
    sb.post_message("SeleniumBase wasn't detected!", duration=3)
```

And for the other site:

```python
from seleniumbase import SB

with SB(uc=True, test=True, incognito=True) as sb:
    url = "https://cfcybernews.eu"
    sb.driver.uc_open_with_reconnect(url, 4)
    if not "没有找到站点" in sb.get_page_title():
        sb.driver.uc_open_with_reconnect(url, 4)

    sb.set_messenger_theme(location="top_center", theme="ice")
    sb.post_message("SeleniumBase wasn't detected!", duration=3)
```

**Comment by OpsecGuy** [comment]
Hey,
I am not sure if I am alone with this issue and what can cause this, but when I build a test script like this:
```
from seleniumbase import Driver

urls = ['https://cfcybernews.eu/', 'https://netguard.io/js_test.php', 'https://gitlab.com/users/sign_in']

for url in urls:
    try:
        driver = Driver(uc=True)
        driver.uc_open_with_reconnect(url, 5.0)
        
        try:
            driver.uc_switch_to_frame("iframe", 1.0)
            driver.uc_click("span.mark")
            driver.switch_to.default_content()
        except Exception:
            pass
        
        driver.sleep(1.5)
        if driver.get_title().lower() == 'just a moment...':
            raise Exception(f'[{url}] Stuck on CF challange')
        
        print(f'[{url}] Closing... CF passed')

    except Exception as err:
        print(err)
    
    finally:
        driver.quit()
```

a Cloudflare JS challenge with Turnstaile captcha stops me all the time and as I said before I am all the time stuck in refreshing page...
Is there anything wrong with my code above? I have tried using various reconnect times and also changing methods of clicking the box inside captcha, but nothing seems to help. It's worth to mention that it started to happen with release of 123 chromedriver. Also I did try using various VPNs and changing internet providers to fix IP reputation, but results are unfortunately very bad ;(
```
[https://cfcybernews.eu/] Stuck on CF challange
[https://netguard.io/js_test.php] Stuck on CF challange
[https://gitlab.com/users/sign_in] Stuck on CF challange
```

In some rare cases when I change IP I can pass through cybernews.eu CF challenge and it doesn't require me to click on captcha due to automatic bypass, but for example, I'm stuck all the time on netguard.io page.

Also, the script in 2nd example you posted doesn't seem to work for me:
![ShareX_vAFGkjE4Vi](https://github.com/seleniumbase/SeleniumBase/assets/30886720/fd244beb-d429-4633-ba7e-386ef5ebfbbd)

-- Disclaimer
I do not advertise any of the above websites. All of them were found by looking for captcha code in HTML from websites like Shodan.io

**Comment by OpsecGuy** [comment]
Problems are gone with the newest update :?

**Reply by mdmintz** [reply]
Is that a question or a statement? Have you tried the newest update?

**Reply by OpsecGuy** [reply]
@mdmintz Indeed I have tried the newest update and all tests on the websites mentioned by me above passed with an additional few to make sure everything is fine :)
